test/extensions_test.dart

```diff
@@ -1352,6 +1353,84 @@ void main() {
         expect(l3.toList(), [4, 5]);
       });
     });
+    group('get frequencies tests', () {
+      test('should return correct frequency map for List of integers', () {
+        var list = [1, 2, 2, 3, 3, 3];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {1: 1, 2: 2, 3: 3});
+      });
+
+      test('should return correct frequency map for List of strings', () {
+        var list = ['a', 'b', 'b', 'c', 'c', 'c'];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {'a': 1, 'b': 2, 'c': 3});
+      });
+
+      test('should handle empty List', () {
+        var list = [];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {});
+      });
+
+      test('should handle single element List', () {
+        var list = [42];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {42: 1});
+      });
+
+      test('should return correct frequency map for Set of integers', () {
+        // ignore: equal_elements_in_set
+        var set = {1, 2, 2, 3, 3, 3};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {1: 1, 2: 1, 3: 1});
+      });
+
+      test('should return correct frequency map for Set of strings', () {
+        // ignore: equal_elements_in_set
+        var set = {'a', 'b', 'b', 'c', 'c', 'c'};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {'a': 1, 'b': 1, 'c': 1});
+      });
+
+      test('should handle empty Set', () {
+        var set = <int>{};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {});
+      });
+
+      test('should handle single element Set', () {
+        var set = {42};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {42: 1});
+      });
+
+      test('should return correct frequency map for Queue of integers', () {
```

Comment:
It's probably overkill to go throguh all kinds of iterables.
I'd rather have more diversity in the elements.

For example having elements that are equal, but not identical, or having elements that are records.
It's the building of the map that is tricky, iterating the iterable will probably work.

Reply:
added `get frequencies tests extended `group